### PR TITLE
Add reliability test about define-undefine network

### PR DESCRIPTION
**Pull request by yalzhang**
Add reliability test about define-start-destroy-undefine network for
1000 times.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
add only 1 case:
virsh.net_define_undefine.normal_test.default_network.reliability
all cases passed:
$ run virsh.net_define_undefine
JOB ID     : ab1c9ea34d12ba822eecb1a9646dd2c5026940f4
JOB LOG    : /root/avocado/job-results/job-2018-09-05T20.05-ab1c9ea/job.log
 (01/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.non_acl: PASS (32.43 s)
 (02/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.acl_test: PASS (38.51 s)
 (03/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_persistent_active: PASS (40.22 s)
 (04/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_transient_active: PASS (35.34 s)
 (05/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.state_persistent_inactive: PASS (35.19 s)
 (06/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.default_network.non_acl: PASS (35.88 s)
 (07/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.default_network.acl_test: PASS (37.50 s)
 (08/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.default_network.reliability: PASS (39.68 s)
 (09/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_network.invalid_uuid: PASS (31.89 s)
 (10/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_network.invalid_netname: PASS (31.59 s)
 (11/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.acl_test.define_acl: PASS (34.16 s)
 (12/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.acl_test.undefine_acl: PASS (34.33 s)
 (13/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.none_option: PASS (30.03 s)
 (14/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.not_exist_option: PASS (31.63 s)
 (15/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.additional_arg: PASS (30.78 s)
 (16/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_define.additional_option: PASS (34.78 s)
 (17/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.none_option: PASS (33.87 s)
 (18/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.not_exist_option: PASS (32.63 s)
 (19/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.additional_arg: PASS (33.70 s)
 (20/20) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.error_test.invalid_options.net_undefine.additional_option: PASS (33.28 s)
RESULTS    : PASS 20 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 695.21 s

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py [hunk 1]
```diff
@@ -123,110 +125,112 @@ def run(test, params, env):
...
-            logging.debug(start_result)
-            start_status = start_result.exit_status
-
-        if trans_ref == "trans":
-            if define_status:
-                fail_flag = 1
-                result_info.append("Define network with right command failed.")
-            else:
-                if start_status:
+        while loop:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: suggest bring try ... catch into the inner of  while loop.
 In this way, any time when issue raised, the finally block can do restore, and the loop can go on further.
Moreover, better logging the loop times after each loop nearly finished.

Comment (yalzhang) [reply libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: updated

File: libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py [hunk 2]
```diff
@@ -205,112 +206,118 @@ def run(test, params, env):
...
+
+                # Undefine network
+                if not check_states:
+                    undefine_result = virsh.net_undefine(undefine_options, undefine_extra,
+                                                         **virsh_dargs)
+                    if trans_ref != "define":
+                        logging.debug(undefine_result)
+                    undefine_status = undefine_result.exit_status
+            except Exception:
+                logging.debug("The define and undefine operation in loop %s failed. " % loop)
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: Others looks fine to me.

[the remark this answers is not in the document]
Comment (yalzhang) [reply libvirt/tests/src/virsh_cmd/network/virsh_net_define_undefine.py]: Updated, Thank you!

**Pull request merged**